app/helpers/character_helper.rb

```diff
@@ -0,0 +1,14 @@
+module CharacterHelper
+  def settings_info(characters)
+    settings = characters.joins(:settings).group(:id)
```

Comment:
Do we know how the SQL is mananging to figure out which table's id to group on?

Reply:
Afaict rails defaults to telling it to use the main table for these, but I can make it explicit if you'd prefer? It is kinda ambiguous xD